consensus/core/src/dag_state.rs

```diff
@@ -275,6 +274,20 @@ impl DagState {
         }
     }
 
+    /// Leader slot of the last commit.
+    pub(crate) fn last_commit_leader(&self) -> Slot {
+        match &self.last_commit {
+            Some(commit) => commit.leader.into(),
+            None => self
+                .genesis
+                .iter()
+                .next()
```

Comment:
Are we sure this is ok to do even with the staked based leader election? Because in this case it won't be the same Slot as the one that gets decided from the leader schedule.

Reply:
The only time `last_commit` will be `None` is at genesis and we filter out leaders from genesis round from the commit sequence. So all genesis blocks are the same as they will never be a leader block.

https://github.com/MystenLabs/sui/blob/main/consensus/core/src/universal_committer.rs#L84-L85